FIX: Wrong sizeof argument of coverity defects(CID 147639)

GeLiXin opened the pull request:
There are 4 defects of wrong size/sizeof argument.
CID 147634, 147635, 147636 seems like false positives.

But 147639 seems really risky:
When array is passed as parameter, it degenerate into pointer,
so the `sizeof(path)` in `is_shorthand_path() `always get return value of 8, not the string lenght we want.

Signed-off-by: GeLiXin <ge.lixin@zte.com.cn>

behlendorf commented on cmd/zpool/zpool_vdev.c:
```diff
@@ -548,7 +548,7 @@ is_shorthand_path(const char *arg, char *path,
 	}
 
-	strlcpy(path, arg, sizeof (path));
+	strlcpy(path, arg, path_size);
```
Yeah, this is clearly wrong and we need to use a passed size.  Similarly you should pass the new `path_size` to `zfs_resolve_shortname()` above and remove the assumption that the buffer is MAXPATHLEN.

behlendorf commented:
@GeLiXin can you incorporate the review feedback and rebase this on master.

GeLiXin commented:
@behlendorf I'm really sorry for feedback for such a long time, as I just come back after a 7 day's holiday of Chinese national day.
I have fixed this PR as review and rebased it now. Please take a review again.

behlendorf commented on cmd/zpool/zpool_vdev.c:
```diff
@@ -660,7 +660,8 @@ make_leaf_vdev(nvlist_t *props, const char *arg, uint64_t is_log)
 	} else {
-		err = is_shorthand_path(arg, path, &statbuf, &wholedisk);
+		err = is_shorthand_path(arg, path, MAXPATHLEN,
+			&statbuf, &wholedisk);
```
[style] Four space indent for continued lines.

```
+		err = is_shorthand_path(arg, path, MAXPATHLEN,
+		    &statbuf, &wholedisk);
```

behlendorf commented:
@GeLiXin no problem.  Hopefully you enjoyed the vacation.  Just one minor style issue to fix and then this LGTM.  @rlaager if you've got a minute it would be great to get another review for this fix.

GeLiXin commented:
@behlendorf  I modified a little to fit different path size which not assumpt it to be MAXPATHLEN any more.
but I didn't find space indent in these two lines you point out, only tabs here.
```
+		err = is_shorthand_path(arg, path, sizeof (path),
+			&statbuf, &wholedisk);
```

behlendorf commented:
@GeLiXin I should have been clearer.  Continued lines should be indented by four spaces after being tab aligned.  I posted the fixed line, you'll need to convert that last tab to four space instead.  You can see this same style used in the rest of the file whenever a line is continued.

The removal of MAXPATHLEN in favor of sizeof is fine.

behlendorf commented on cmd/zpool/zpool_vdev.c:
```diff
@@ -658,9 +658,10 @@ make_leaf_vdev(nvlist_t *props, const char *arg, uint64_t is_log)
 	} else {
-		err = is_shorthand_path(arg, path, &statbuf, &wholedisk);
+		err = is_shorthand_path(arg, path, sizeof (path),
+			&statbuf, &wholedisk);
```
[style]

GeLiXin commented:
@behlendorf I get your point now, thanks.  And I'm done!
